### ISSUE TITLE
Add Possibility to edit profile

### ISSUE THREAD

**Issue by switch87** [WelcometoMyGarden/welcometomygarden#76]
At the moment there is no button to edit your account name or email address.

**Comment by MichielLeyman** [comment]
Agreed that this is much needed, especially so that we can more easily comply with people wanting to update their data.

Updating the name is quite simple, but I've been dancing around the issue of updateability of the email address. Firebase will send a default mail template to a user when their email is changed, allowing them to revert that change. That's cool, but it's in Firebase's default template, not in the same style as our other emails.

Right now in the Admin SDK we can use `generatePasswordResetLink` and `generateEmailVerificationLink`, allowing us to completely roll our own solution for emails. However, it's not possible for us to generate the custom link for email recovery, nor is it possible to prevent the email being sent when the function is called. Meaning that we either bypass it being sent with `updateUser` instead of using `updateEmail`, or roll our own solution entirely.

**Comment by MichielLeyman** [comment]
Before having our own API, changing emails, deleting accounts, changing names, ... turned out to be more complex than it was worth. No longer! This will require some design input in the front-end, as well as special attention to delete cascade logic in the backend. 

What happens to a deleted user's chats?

- [ ] User deletion
- [ ] Update of first - and last name
- [ ] Email update: send a notification to the old email that the new one was updated as well + either reverify password or reverify email

**Comment by th0rgall** [comment]
Changing your own email address is now possible! https://github.com/WelcometoMyGarden/welcometomygarden/pull/321

When requesting a change, a verification email is sent to the new email. The change only gets applied after the user clicks that link. After the change, a default Firebase recovery email is sent to the old email, in case they want to undo the change. 

![CleanShot 2023-05-05 at 16 53 36](https://user-images.githubusercontent.com/11543641/236477416-faf503e7-eba7-4ce3-817a-ffb6b613ca95.png)
![CleanShot 2023-05-05 at 16 53 52](https://user-images.githubusercontent.com/11543641/236477276-7ddb19f4-2ccc-47ba-a68a-80f43bb40954.png)

Changing your account name is not yet possible, we're getting this request much less often.
